From 01ae1f06ad32950496627be2842eca4d0976626e Mon Sep 17 00:00:00 2001
From: Herrie <Github.com@herrie.org>
Date: Wed, 20 Sep 2023 11:07:14 +0200
Subject: [PATCH] com.webos.nodejs.role.json: Add permissions required for
 legacy services

Solves errors like:

Can not find service "org.webosports.service.update" permissions for executable "/usr/bin/node"

Signed-off-by: Herman van Hazendonk <github.com@herrie.org>
---
Upstream-Status: Inappropriate [LuneOS specific]

 files/sysbus/com.webos.nodejs.role.json.in | 66 ++++++++++++++++++++++
 1 file changed, 66 insertions(+)

diff --git a/files/sysbus/com.webos.nodejs.role.json.in b/files/sysbus/com.webos.nodejs.role.json.in
index 765cca6..f92d062 100755
--- a/files/sysbus/com.webos.nodejs.role.json.in
+++ b/files/sysbus/com.webos.nodejs.role.json.in
@@ -6,7 +6,73 @@
     "permissions": [
         {
             "service": "*",
+            "inbound": [],
             "outbound": []
+        },
+        {
+            "service": "com.palm.keymanager",
+            "inbound": ["*"],
+            "outbound": ["*"]
+        },
+        {
+            "service": "com.palm.mediapermissions",
+            "inbound": ["*"],
+            "outbound": ["*"]
+        },
+        {
+            "service": "com.palm.service.accounts",
+            "inbound": ["*"],
+            "outbound": ["*"]
+        },
+        {
+            "service": "com.palm.service.calendar.reminders",
+            "inbound": ["*"],
+            "outbound": ["*"]
+        },
+        {
+            "service": "com.palm.service.contacts",
+            "inbound": ["*"],
+            "outbound": ["*"]
+        },
+        {
+            "service": "com.palm.service.contacts.linker",
+            "inbound": ["*"],
+            "outbound": ["*"]
+        },
+        {
+            "service": "org.webosports.service.tweaks.prefs",
+            "inbound": ["*"],
+            "outbound": ["*"]
+        },
+        {
+            "service": "org.webosports.service.cdav",
+            "inbound": ["*"],
+            "outbound": ["*"]
+        },
+        {
+            "service": "org.webosports.service.devmode",
+            "inbound": ["*"],
+            "outbound": ["*"]
+        },
+        {
+            "service": "org.webosports.service.filemanager",
+            "inbound": ["*"],
+            "outbound": ["*"]
+        },
+        {
+            "service": "org.webosports.service.licenses",
+            "inbound": ["*"],
+            "outbound": ["*"]
+        },
+        {
+            "service": "org.webosports.service.update",
+            "inbound": ["*"],
+            "outbound": ["*"]
+        },
+        {
+            "service": "",
+            "inbound": ["*"],
+            "outbound": ["*"]
         }
     ]
 }
